core/jvm/src/test/scala/scalaz/zio/QueueSpec.scala

```diff
@@ -152,16 +152,16 @@ class QueueSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRun
       values = Range.inclusive(1, 10).toList
       f      <- IO.forkAll(values.map(queue.offer))
       _      <- waitForSize(queue, 10)
-      l      <- queue.take.repeat(ZSchedule.recurs(9) *> ZSchedule.identity[Int].collect)
+      l      <- queue.take.repeat(ZSchedule.recurs(9) *> ZSchedule.identity[Int].collect.immediately)
       _      <- f.join
     } yield l must containTheSameElementsAs(values))
 
   def e5 =
     unsafeRun((for {
       queue        <- Queue.bounded[Int](10)
-      _            <- queue.offer(1).repeat(ZSchedule.recurs(9))
+      _            <- queue.offer(1).repeat(ZSchedule.recurs(9).immediately)
```

Comment:
All the `immediately` everywhere concern me... why do we need to ignore the initial delay?

Reply:
Actually `immediately` is giving an initial delay of `zero` to an existent schedule. I thought about it to avoid breaking current schedule implementations.

Reply:
I think we should encourage the. new pattern in the ZIO code base, i.e. do not run the first one until it's specified to run. 

What does `retry` do with the initial delay? I suppose it would ignore it, right?

Reply:
I agree. I just got shy to change signatures :smile: 
But make sense changing tthem

On my last commit I made `retry` considering initial delay like `repeat`